qubesbuilder/plugins/source_deb/scripts/clamp-changelog-entry-date

```diff
@@ -30,9 +31,14 @@ if [ ! -r "$CHANGELOG_PATH" ]; then
     exit 1
 fi
 
-# get previous date
-PREVIOUS_DATE=$(grep '^ --' "$CHANGELOG_PATH" | head -n 2 | tail -n 1 | grep -o '  .*')
+# Set CURRENT_DATE based on SOURCE_DATE_EPOCH or fallback to previous date
+if [ -n "$SOURCE_DATE_EPOCH" ]; then
+    CURRENT_DATE=$(date -R -u -d "@$SOURCE_DATE_EPOCH")  # RFC 2822 format (e.g., Wed, 02 Oct 2002 15:00:00 +0200)
+else
+    # Get previous date from changelog
+    CURRENT_DATE=$(grep '^ --' "$CHANGELOG_PATH" | head -n 2 | tail -n 1 | grep -o '  .*')
+fi
```

Comment:
To be honest, I don't think this is a good idea. The reference should be the date provided by the CHANGELOG which means that SOURCE_DATE_EPOCH should somehow match that on Debian side.

Reply:
I see your point about using the SOURCE_DATE_EPOCH as the reference for Debian reproducibility. I've look adjusted the script so that SOURCE_DATE_EPOCH is used more effectively, ensuring consistency with Debian's side rather than relying on the previous changelog entry date.